### PR TITLE
rsx: clipping / misc

### PR DESCRIPTION
**Pull request by jarveson**
Changes/fixes:
- rsx: Fix depth clipping
-- Should now correctly scale/offset z for viewport, which may help in random clipping issues. tested game doesnt work on master without other hacks (Vancouver 2010) but this fixes graphics display in it
-- (comment in the code before mentions 'Ni no Kuni', if someone could test that graphics don't regress on it, that would be great!)
- rsx: fix image_in arg decoding / swizzle
-- Fixes x/y decoding, fixes some psl1ght blitting samples
-- After seeing countless weird values games seem to give for swizzle size, it seems the rsx just ignores the values, fixes cpu blit in age of booty
- d3d12 crash fix

### CODE REVIEW

**Comment by kd-11** [comment]
minor concerns about blit engine coordinates, but the rest is fine. I seem to recall blit engine copying tiled sections out to display buffer and it worked correctly - you can check this by disabling the render target check in flip code and only using the texture cache path instead (on opengl) to flip the actual display buffer.

**Comment by digitaldude555** [comment]
Master
screenshot 812

Your build
screenshot 811

**Comment by jarveson** [comment]
fixed, also fixed force cpu blit with that game

File: rpcs3/Emu/RSX/rsx_decode.h [hunk 1]
```diff
@@ -4064,21 +4078,23 @@ struct registers_decoder<NV3089_IMAGE_IN>
 	public:
 		decoded_type(u32 raw_value) { m_data.raw_value = raw_value; }
 
-		u16 x() const
+		// x and y given as 16 bit fixed point
+
+		f32 x() const
 		{
-			return m_data.x;
+			return m_data.x / 16.f;
```
Comment (Marthgun) [inline rpcs3/Emu/RSX/rsx_decode.h]: Why divide by 16.0 here?  i can't follow it.

Comment (jarveson) [reply rpcs3/Emu/RSX/rsx_decode.h]: its a 16 bit fixed point number, so it needs to be divided to turn back into the correct float, also confirmed in psl1ght sources https://github.com/ps3dev/PSL1GHT/blob/534e58950732c54dc6a553910b653c99ba6e9edc/samples/graphics/blitting/source/main.c#L180

Comment (TGEnigma) [reply rpcs3/Emu/RSX/rsx_decode.h]: Maybe make a function for it? it'd be better if it were explicit

**Comment by kd-11** [comment]
I'll patch the depth clipping emulation for gl and vulkan later as well as reimplementing the remaining z manipulation bits like polygon offset and removing 'magic' paths in texture cache related to this fix. Looks good as is.

**Pull request merged**